euth/dashboard/views.py

```diff
@@ -234,6 +240,34 @@ def get_form_kwargs(self):
         return kwargs
 
 
+class DashboardProjectArchiveView(DashboardBaseMixin,
+                                  rules_views.PermissionRequiredMixin,
+                                  generic.UpdateView):
+    model = project_models.Project
+    form_class = forms.ProjectArchiveForm
+    success_message = _('Project has been archived.')
+    permission_required = 'euth_organisations.initiate_project'
+    slug_url_kwarg = 'project_slug'
+    menu_item = 'project'
+    archiving = False
+
+    @property
+    def raise_exception(self):
+        return self.request.user.is_authenticated()
+
+    def get_success_url(self):
+        return reverse('dashboard-project-list',
+                       kwargs={
+                           'organisation_slug': self.organisation.slug,
+                       })
+
+    def form_valid(self, form):
+        project = form.instance
+        project.is_archived = self.archiving and project.has_finished
```

Comment:
> I am wondering whether it wouldn't be better to validate in form.clean that the project has indeed finished

I like the idea. Since the `form.clean` requires raising an exception and the current templates do not take exceptions into account, I would prefer `project.is_archived = self.archiving and project.has_finished`.